spec/models/page_redirect_spec.rb

```diff
@@ -0,0 +1,29 @@
+require "rails_helper"
+
+RSpec.describe PageRedirect, type: :model do
+  subject { build(:page_redirect) }
+
+  describe "validations" do
+    it { is_expected.to validate_uniqueness_of(:old_path) }
+    it { is_expected.to validate_presence_of(:old_path) }
+    it { is_expected.to validate_presence_of(:new_path) }
+    it { is_expected.to validate_presence_of(:source) }
+    it { is_expected.to validate_inclusion_of(:source).in_array(described_class::SOURCES) }
```

Comment:
the trick with this is that it will always be true no matter how many times we change `SOURCES`. I tend to keep the names hardcoded here, so that these checks will intentionally fail if I didn't mean to change it or forgot to update them

Reply:
That's fair. This spec is really just making sure it's something we validate, so it would fail if someone removed the validation from the model. So we could add another spec to test the options or take your suggestion. I took your suggestion 💯